### PR TITLE
Bloctans changes chat

### PR DESCRIPTION
**Pull request by Bloctans**
cst, can you do:
~~-inbox~~ Done by Tnix
-creating chats
-leaving chats
-reloading
~~and load more on list and chats please? thx~~ Done by Tnix
im too lazy to continue this and dont want to

### CODE REVIEW

File: src/screens/Setup.svelte [hunk 1]
```diff
@@ -115,13 +115,13 @@
 				} else if (code == "I:011 | Invalid Password") {
 					loginStatus = "Invalid password!";
 				} else if (code == "E:018 | Account Banned") {
-					loginStatus = "This account is banned. L :(";
+					loginStatus = "This account is banned. LOL :(";
```
Comment (tnix100) [inline src/screens/Setup.svelte:118]: I liked "L" better

File: src/screens/Setup.svelte [hunk 2]
```diff
@@ -115,13 +115,13 @@
...
 				} else if (code == "E:107 | Packet too large") {
-					loginStatus = "The username and/or password is too long!";
+					loginStatus = "The username and/or password is too long! Sadly";
```
Comment (tnix100) [inline src/screens/Setup.svelte:120]: Sadly, the "Sadly" doesn't really fit

File: src/screens/Setup.svelte [hunk 3]
```diff
@@ -115,13 +115,13 @@
...
 				} else if (code == "E:019 | Illegal characters detected") {
 					loginStatus = "Usernames must not have spaces or other special characters!";
 				} else if (code == "E:106 | Too many requests") {
-					loginStatus = "Too many requests! Please try again later.";
+					loginStatus = "Too many requests! Please try again later. And While your at it blame Us for it";
```
Comment (tnix100) [inline src/screens/Setup.svelte:124]: BIt of grammar and capitalization error. It should be, "And while you're at it, blame us for it."
But still don't think it really fits with the error.

File: src/lib/Post.svelte [hunk 4]
```diff
@@ -72,22 +77,42 @@
 		<button 
 			class="pfp" 
 			on:click={()=>{
-				profileClicked.set(post.user);
+				if (post.user == "Discord") {
```
Comment (tnix100) [inline src/lib/Post.svelte:80]: why don't you also set `post.user` to the bridged user?

File: src/screens/Sidebar.svelte [hunk 5]
```diff
@@ -18,11 +19,23 @@
 	import settings from "../assets/settings.svg";
 	import logout from "../assets/logout.svg";
 	import groupcat from "../assets/meowy.svg";
+	import gc from "../assets/chat.svg"
 
 	/**
 	* @param {any} newPage Goes to a page while also refreshing it.
 	*/
 	function goto(newPage) {
+		cljs.send({
```
Comment (tnix100) [inline src/screens/Sidebar.svelte:28]: uhh, don't send a request to leave the group chat everytime a page is switched, you should check if the last page was a group chat or not

File: src/screens/Setup.svelte (hunk 1 above)
Comment (Bloctans) [reply src/screens/Setup.svelte:118]: ok will change

File: src/lib/Post.svelte (hunk 4 above)
Comment (Bloctans) [reply src/lib/Post.svelte:80]: ?

File: src/screens/Sidebar.svelte (hunk 5 above)
Comment (Bloctans) [reply src/screens/Sidebar.svelte:28]: uhhh how could i do that quickly

File: src/lib/Post.svelte (hunk 4 above)
Comment (tnix100) [reply src/lib/Post.svelte:80]: In the other part of showing the post where it sets `userName` to the bridged user, why not also set `post.user` to it?

Comment (Bloctans) [reply src/lib/Post.svelte]: yeah im stupid for not knowing

File: src/screens/Sidebar.svelte (hunk 5 above)
Comment (tnix100) [reply src/screens/Sidebar.svelte:28]: > uhhh how could i do that quickly

https://github.com/meower-media-co/Meower-Svelte/pull/20

**Comment by tnix100** [comment]
please review this for the group chat thing https://github.com/meower-media-co/Meower-Svelte/pull/20

**Comment by tnix100** [comment]
Now we wait for @CST1229 to review, but it seems good to me!

**Comment by Bloctans** [comment]
ok now we can wait for review, had a lil messup

File: src/lib/Post.svelte [hunk 6]
```diff
@@ -23,10 +23,10 @@
 		var userName = ""
 		if (post.user == "Discord") {
 			post.user = post.content.split(":")[0];
+			post.content = post.content.slice(post.content.indexOf(":")+2);
```
Comment (tnix100) [inline src/lib/Post.svelte:26]: would this work a bit better?
image

Comment (Bloctans) [reply src/lib/Post.svelte:26]: actually yeah

Comment (Bloctans) [reply src/lib/Post.svelte:26]: for post.content? no. because if i did this: hi: hi on discord, it would render as hi

Comment (tnix100) [reply src/lib/Post.svelte:26]: wait, it would be `[1]` not `[0]`, whoops

**Comment by tnix100** [comment]
@Bloctans https://github.com/meower-media-co/Meower-Svelte/pull/21

**Comment by tnix100** [comment]
I found some bugs, I will fix them and add inbox messages, then we can merge this.

**Comment by tnix100** [comment]
@CST1229 can we merge now?

**Comment by tnix100** [comment]
There is one small bug, I will fix soon

**Comment by Bloctans** [comment]
todo:
- add members to chat
- remove "add to chat" functionallity
- remove members from chat
- view member profiles
- leave chats
- join chats

**Comment by Bloctans** [comment]
Note to CST: we are still working on it

**Comment by tnix100** [comment]
Moved to https://github.com/meower-media-co/Meower-Svelte/pull/24

**Pull request closed without merging**